pkg/asset/installconfig/gcp/validation_test.go

```diff
@@ -122,6 +149,30 @@ func TestGCPInstallConfigValidation(t *testing.T) {
 			expectedError:  true,
 			expectedErrMsg: "computeSubnet: Invalid value.*controlPlaneSubnet: Invalid value",
 		},
+		{
+			name:           "Valid machine types",
+			edits:          editFunctions{validMachineTypes},
+			expectedError:  false,
+			expectedErrMsg: "",
+		},
+		{
+			name:           "Invalid default machine type",
+			edits:          editFunctions{invalidateDefaultMachineTypes},
+			expectedError:  true,
+			expectedErrMsg: "[platform.gcp.defaultMachinePlatform.type: Invalid value: \"n1-standard-1\": instance type does not meet minimum resource requirements of 4 vCPUs, platform.gcp.defaultMachinePlatform.type: Invalid value: \"n1-standard-1\": instance type does not meet minimum resource requirements of 15360 MB Memory]",
+		},
+		{
+			name:           "Invalid control plane machine types",
+			edits:          editFunctions{invalidateControlPlaneMachineTypes},
+			expectedError:  true,
+			expectedErrMsg: "[controlPlane.platform.gcp.type: Invalid value: \"n1-standard-1\": instance type does not meet minimum resource requirements of 4 vCPUs, controlPlane.platform.gcp.type: Invalid value: \"n1-standard-1\": instance type does not meet minimum resource requirements of 15360 MB Memory]",
+		},
+		{
+			name:           "Invalid compute machine types",
+			edits:          editFunctions{invalidateComputeMachineTypes},
+			expectedError:  true,
+			expectedErrMsg: "[compute[0].platform.gcp.type: Invalid value: \"n1-standard-1\": instance type does not meet minimum resource requirements of 2 vCPUs, compute[0].platform.gcp.type: Invalid value: \"n1-standard-1\": instance type does not meet minimum resource requirements of 7680 MB Memory]",
+		},
```

Comment:
Can you add a test case where the instance type selected is not a valid instance type?

Reply:
Done... https://github.com/openshift/installer/commit/0d74861bd7fa7cb28cde3f79fa63dc43704b69b2